src/test/java/com/williamfiset/algorithms/graphtheory/TravelingSalesmanProblemTest.java

```diff
@@ -11,58 +11,85 @@
 
   @Test(expected = IllegalArgumentException.class)
   public void testTspRecursiveInvalidStartNode() {
-    double[][] dist = {
-      {1, 2, 3},
-      {4, 5, 6},
-      {7, 8, 9}
-    };
-    new TspDynamicProgrammingRecursive(321, dist);
+    new TspDynamicProgrammingRecursive(321, new double[5][5]);
   }
 
   @Test(expected = IllegalArgumentException.class)
-  public void testTspIterativeInvalidStartNode() {
-    double[][] dist = {
-      {1, 2, 3},
-      {4, 5, 6},
-      {7, 8, 9}
-    };
-    new TspDynamicProgrammingIterative(321, dist);
+  public void testTspIterativeStartNodeTooLarge() {
+    new TspDynamicProgrammingIterative(321, new double[5][5]);
   }
 
-  @Test(expected = IllegalStateException.class)
+  @Test(expected = IllegalArgumentException.class)
+  public void testTspIterativeNegativeStartNode() {
+    new TspDynamicProgrammingIterative(-1, new double[5][5]);
+  }
+
+  @Test(expected = IllegalArgumentException.class)
   public void testTspRecursiveNonSquareMatrix() {
-    double[][] dist = {
-      {1, 2, 3},
-      {4, 5, 6}
-    };
-    new TspDynamicProgrammingRecursive(dist);
+    new TspDynamicProgrammingRecursive(new double[5][4]);
   }
 
   @Test(expected = IllegalStateException.class)
   public void testTspIterativeNonSquareMatrix() {
-    double[][] dist = {
-      {1, 2, 3},
-      {4, 5, 6}
-    };
-    new TspDynamicProgrammingIterative(dist);
+    new TspDynamicProgrammingIterative(new double[5][4]);
   }
 
   @Test(expected = IllegalStateException.class)
   public void testTspRecursiveSmallGraph() {
-    double[][] dist = {
-      {0, 1},
-      {1, 0}
-    };
-    new TspDynamicProgrammingRecursive(dist);
+    new TspDynamicProgrammingRecursive(new double[2][2]);
   }
 
   @Test(expected = IllegalStateException.class)
   public void testTspIterativeSmallGraph() {
-    double[][] dist = {
-      {0, 1},
-      {1, 0}
-    };
-    new TspDynamicProgrammingIterative(dist);
+    new TspDynamicProgrammingIterative(new double[2][2]);
+  }
+
+  @Test(expected = IllegalArgumentException.class)
+  public void testTspIterativeTooLargeGraph() {
+    new TspDynamicProgrammingIterative(new double[33][33]);
+  }
+
+  /* Test that the cached execution time of getTour() is less than 20% of the uncached run time.
```

Comment:
Why are you adding tests to verify the cached vs uncached time? That's not a functional thing that needs to be tested

Reply:
That is true, but I wanted to get 100% branch coverage which needed the caching to be tested, but this isn't completely necessary like you're saying so it could be removed.

Reply:
It also felt at least a little bit logical to make sure the caching actually does work, and that the tour isn't calculated a second time.